server.go

```diff
@@ -571,6 +571,7 @@ func (srv *Server) serveDNS(m []byte, w *response) {
 		req.SetRcodeFormatError(req)
 		// Are we allowed to delete any OPT records here?
 		req.Ns, req.Answer, req.Extra = nil, nil, nil
+		req.Zero = false
```

Comment:
while this is not currently explicitly tested, it seems like a good default when rejecting the message.